src/Utils.js

```diff
@@ -19,6 +19,10 @@ export function internalInstance(inst) {
     inst[internalInstanceKey(inst)];
 }
 
+export function isFunctionalComponent(inst) {
+  return inst.constructor.name === 'StatelessComponent';
```

Comment:
Provided what is passed to the function is an instance, yes, I think it will have a constructor.